src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -22,85 +24,157 @@ internal abstract class AbstractRemoveUnusedVariableCodeFixProvider<TLocalDeclar
     {
         protected abstract bool IsCatchDeclarationIdentifier(SyntaxToken token);
 
-        public async override Task RegisterCodeFixesAsync(CodeFixContext context)
-        {
-            foreach (var diagnostic in context.Diagnostics)
-            {
-                var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
-                var token = root.FindToken(diagnostic.Location.SourceSpan.Start);
-                if (!IsCatchDeclarationIdentifier(token))
-                {
-                    var ancestor = token.GetAncestor<TLocalDeclarationStatement>();
+        protected abstract SyntaxNode GetNodeToRemoveOrReplace(SyntaxNode node);
 
-                    if (ancestor == null)
-                    {
-                        return;
-                    }
-                }
-            }
+        protected abstract void RemoveOrReplaceNode(SyntaxEditor editor, SyntaxNode node, ISyntaxFactsService syntaxFacts);
+
+        protected abstract SeparatedSyntaxList<SyntaxNode> GetVariables(TLocalDeclarationStatement localDeclarationStatement);
 
-            context.RegisterCodeFix(
-                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
-                context.Diagnostics);
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Diagnostic diagnostic = context.Diagnostics.Single();
+            context.RegisterCodeFix(new MyCodeAction(async c => await FixAsync(context.Document, diagnostic, c).ConfigureAwait(false)), diagnostic);
+            return Task.CompletedTask;
         }
 
-        protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor syntaxEditor, CancellationToken cancellationToken)
```

Comment:
can you keep the name 'editor' here?  it will make it easier to review.

Reply:
I'd also prefer to keep the name for 'editor'. I have to rename it to be consistent with 'documentEditor'. 
There is a 'documentEditor' introduced in the method. Having one of them as 'editor' and another as 'documentEditor' looks confusing. So, I suggest to keep names consistent within the method: 'syntaxEditor' and 'documentEditor'.


---
In reply to: [160015903](https://github.com/dotnet/roslyn/pull/24077#discussion_r160015903) [](ancestors = 160015903)